internal/app/fetcher/libraries.go

```diff
@@ -0,0 +1,77 @@
+package fetcher
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/xssnick/tonutils-go/tlb"
+	"github.com/xssnick/tonutils-go/tvm/cell"
+)
+
+type LibDescription struct {
+	_          tlb.Magic        `tlb:"$00"`
+	Lib        *cell.Cell       `tlb:"^"`
+	Publishers *cell.Dictionary `tlb:"dict inline 256"`
+}
+
+func (s *Service) GetAccountLibraries(ctx context.Context, raw *tlb.Account) (*cell.Cell, error) {
+	hashes, err := findLibraries(raw.Code)
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "find libraries")
+	}
+
+	libs, err := s.API.GetLibraries(ctx, hashes...)
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "get libraries")
+	}
+
+	libsMap := cell.NewDict(256)
+
+	for i, hash := range hashes {
+		desc := LibDescription{Lib: libs[i]}
+
+		h := cell.BeginCell().MustStoreSlice(hash, 256).EndCell()
+		t, err := tlb.ToCell(desc)
+
+		if err != nil {
+			return nil, err
+		}
+
+		err = libsMap.Set(h, t)
+		s.libraries.set(hash, &desc)
+
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return libsMap.ToCell()
+}
+
+func getLibraryHash(code *cell.Cell) ([]byte, error) {
+	hash, err := code.BeginParse().LoadBinarySnake()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return hash[1:], nil
+}
+
+// TODO recursive Refs
```

Comment:
Looks like it is not hard to implement using `PeekRef` of `tonutils-go` library.

Reference:
https://github.com/tonkeeper/tongo/blob/c6497b2bfc8d49688be1c9b3201806cd1522a8c1/txemulator/libraries.go#L26